rest_sensors.cpp

```diff
@@ -799,6 +799,39 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                 else if (rid.suffix == RConfigTempThreshold || rid.suffix == RConfigHumiThreshold)
                 {
                 }
+                else if (rid.suffix == RConfigLock)
+                {
+                    bool ok;
+                    
+                    item = sensor->item(RConfigLock);
+                    if (item && map[pi.key()].type() == QVariant::Bool)
```

Comment:
At this point `item` is already set correctly via `item = sensor->item(rid.suffix)` above and checked for `nullptr`.

So `item = sensor->item(RConfigLock);` should be removed.

```
if (map[pi.key()].type() == QVariant::Bool)
{
 ...
}
else
{
    rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/sensors/%1/config/lock").arg(id), QString("invalid value, %1, for parameter, lock").arg(val.toString())));
    rsp.httpStatus = HttpStatusBadRequest;
    return REQ_READY_SEND;
}
```

The else branch to raise ERR_INVALID_VALUE is missing at the end.

Reply:
done